broke-broker/src/main/java/org/dsngroup/broke/broker/storage/ServerSessionCollection.java

```diff
@@ -0,0 +1,62 @@
+package org.dsngroup.broke.broker.storage;
+
+import io.netty.handler.codec.mqtt.MqttMessage;
+
+import java.util.HashMap;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * The singleton class ServerSessionCollection
+ * Contain all the Server Sessions for all clients
+ * */
+public class ServerSessionCollection {
+
+    // Use volatile to guarantee all session user threads gets the well initialized instance.
+    private volatile static ServerSessionCollection serverSessionCollection;
+
+    private HashMap<String, ServerSession> serverSessionCollectionMap;
+
+    /**
+     * Getter method of the server session collection
+     * */
+    public static ServerSessionCollection getServerSessionCollection() {
+        if(serverSessionCollection==null) {
+            synchronized (ServerSessionCollection.class) {
+                if(serverSessionCollection==null) {
+                    serverSessionCollection = new ServerSessionCollection();
```

Comment:
Weird here.

Reply:
This is how multi-thread singleton pattern avoids instantiating more than once.

Reply:
Ok, I got it.